extensions/google-translate/src/instant-translate.tsx

```diff
@@ -0,0 +1,84 @@
+import { getPreferenceValues, getSelectedText, showHUD, Clipboard } from "@raycast/api";
+import { simpleTranslate } from "./simple-translate";
+import { LanguageCode } from "./languages";
+
+// Helper function to show HUD for a longer duration based on text length
+async function showExtendedHUD(message: string, minDurationMs = 2000) {
+  await showHUD(message);
+
+  // Calculate duration based on message length
+  // Assuming average reading speed of 150ms per character (adjust as needed)
+  const readingTimePerChar = 150; // milliseconds
+  const calculatedDuration = Math.max(minDurationMs, message.length * readingTimePerChar);
+
+  // Cap maximum duration to avoid excessive waiting
+  const maxDuration = 15000; // 15 seconds max
+  const finalDuration = Math.min(calculatedDuration, maxDuration);
+
+  // Show the same message multiple times with a delay to extend visibility
+  const intervalTime = 1000; // Refresh every 1 second
+  const iterations = Math.floor(finalDuration / intervalTime);
+
+  for (let i = 0; i < iterations; i++) {
+    await new Promise((resolve) => setTimeout(resolve, intervalTime));
+    await showHUD(message);
+  }
+}
+
+export default async function InstantTranslate() {
+  try {
+    const preferences = getPreferenceValues<ExtensionPreferences>();
+    const targetLanguage = preferences.lang2; // Use secondary language as target
+    const sourceLanguage = preferences.lang1 === "auto" ? "auto" : preferences.lang1;
+    const defaultAction = preferences.defaultAction; // Can be "copy", "paste", or "show"
```

Comment:
don't need this comment, this property is typed with typescript, so there will be shown possible values in intelisens

<img width="862" alt="Screenshot 2025-06-27 at 16 41 13" src="https://github.com/user-attachments/assets/b216245d-8f03-489c-98a7-35b102411ad4" />